strapi/strapi
# Changing locale is now stored in db

### What does it do?

Allows to store preferred language in db


### How to test it?

The behavior I'm not sure how to manage is the first time a user will use this functionality. At that time, nothing is stored in the backend, so we're not able to retrieve the preferred locale for the user. At this point, I've kept the same behavior as before.

Here are some steps and scenarios to check (let me know if some are missing):

- Logout from the app
- Clear your local storage
- Refresh the page
- **You should be in your browser lang by default**
- Login with your username
- Check your local storage: **nothing you should be there concerning the app language**
- Click the user dropdown on the top right of the screen and select "Profil"
- Select "profile".
- At this point, since no preferred languages have been chosen, **the language select should be empty**
- Select a language, like "English"
- Click the save button at the top right of the page
- **The language of the app should have changed**
- **The locale storage value should have changed accordingly**
- Logout again
- **The language used in the login screen should be the same as the one selected on the profile page**
- Login again
- **The user interface should be translated in the language previously selected**
- Click the user dropdown on the top right of the screen and select "Profil"
- Select "profile".
- **The language select should be filled with the previously selected language**

### Question concerning the login page

Since we don't have access to the user at this point, I'm only relying on the user locale storage to manage this scenario. But what should happen when the user changes locale on the login screen and then logs in to the application?
